src/segment_git.go

```diff
@@ -131,12 +135,23 @@ func (g *git) string() string {
 	if g.props.getBool(StatusColorsEnabled, false) {
 		g.SetStatusColor()
 	}
+	enableHyperlink := g.props.getBool(EnableHyperlink, false)
 	buffer := new(bytes.Buffer)
+
+	// url
+	upstream := replaceAllString("/.*", g.repo.upstream, "")
+	g.repo.url = g.getGitCommandOutput("remote", "get-url", upstream)
+
 	// branchName
 	if g.repo.upstream != "" && g.props.getBool(DisplayUpstreamIcon, false) {
-		fmt.Fprintf(buffer, "%s", g.getUpstreamSymbol())
+		fmt.Fprintf(buffer, "%s", g.getUpstreamSymbol(g.repo.url))
+	}
+	if g.repo.upstream != "" && enableHyperlink {
+		url := strings.Replace(g.repo.url, ".git", "", -1)
+		fmt.Fprintf(buffer, "]8;;%s\\%s]8;;\\", fmt.Sprintf("%s/tree/%s", url, g.repo.HEAD), g.repo.PRETTYHEAD)
```

Comment:
The `/tree/` url seems github specific. At my work, we use Azure Devops and the url is different there. 

Example:
git repo url:
`https://AcmeSoftware@dev.azure.com/AcmeSoftware/OhMyPosh/_git/oh-my-posh3`
branch:
`feat/hyperlink`
url:
`https://AcmeSoftware@dev.azure.com/AcmeSoftware/OhMyPosh/_git/oh-my-posh3?version=GBfeat%2Fmy-feature`

 The url consruction code would be something like this:
```go
fmt.Sprintf("%s?version=GB%s", url, g.repo.HEAD)
```

Maybe we could do a check to see if the url contains `github.com` or `dev.azure.com` and apply the proper url building? And for other domains we don't know about, just omit the hyperlink feature. Or make the hyperlink point to the git repo url directly (no branch)

Reply:
I was expecting some issues like that. Yeah of course let's keep a list of main providers and point to the repo when unknown.
Thanks for the feedback!

Reply:
I did a first draft of multi providers support, if you have the opportunity to test it.